prowler/providers/kubernetes/kubernetes_provider.py

```diff
@@ -0,0 +1,72 @@
+import os
+import sys
+
+from kubernetes import client, config
+
+from prowler.lib.logger import logger
+
+
+class Kubernetes_Provider:
+    def __init__(
+        self,
+        kubeconfig_file: str,
+        context: list,
+    ):
+        logger.info("Instantiating Kubernetes Provider ...")
+        self.api_client, self.context = self.__set_credentials__(
+            kubeconfig_file, context
+        )
+        if not self.api_client:
+            logger.critical("Failed to set up a Kubernetes session.")
+            sys.exit(1)
+
+    def __set_credentials__(self, kubeconfig_file, context):
+        try:
+            if kubeconfig_file:
+                # Use kubeconfig file if provided
+                config.load_kube_config(
+                    config_file=os.path.abspath(kubeconfig_file), context=context
+                )
+            else:
+                # Otherwise try to load in-cluster config
+                config.load_incluster_config()
+            context = config.list_kube_config_contexts()[0][0]
+            return client.ApiClient(), context
+        except Exception as error:
+            logger.critical(
+                f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+            sys.exit(1)
+
+    def get_credentials(self):
+        return self.api_client, self.context
+
+    def get_context_user_roles(self):
+        try:
+            rbac_api = client.RbacAuthorizationV1Api()
+            context_user = self.context.get("context").get("user")
+            roles = []
+
+            # Search in ClusterRoleBindings
+            cluster_role_bindings = rbac_api.list_cluster_role_binding()
+            for crb in cluster_role_bindings.items:
+                if crb.subjects:
+                    for subject in crb.subjects:
+                        if subject.kind == "User" and subject.name == context_user:
+                            roles.append(f"ClusterRole: {crb.role_ref.name}")
+
+            # Search in RoleBindings for all namespaces
+            role_bindings = rbac_api.list_role_binding_for_all_namespaces()
+            for rb in role_bindings.items:
+                for subject in rb.subjects:
+                    if subject.kind == "User" and subject.name == context_user:
+                        roles.append(
+                            f"Role ({rb.metadata.namespace}): {rb.role_ref.name}"
+                        )
```

Comment:
I think you can merge this two into their own function like 
```python
roles = search_roles(rbac_api.list_cluster_role_binding())
roles = search_roles(rbac_api.list_role_binding_for_all_namespaces())

def search_roles(role_bindings):
    for rb in role_bindings.items:
        if rb.subjects:
              for subject rb.subjects:
                    .....
```

Reply:
But each case has a different kind of append...

Reply:
But I assume you can check the difference within the loop.

Reply:
like:

```python
roles = []
# Search in ClusterRoleBindings
roles = search_and_save_roles(
    roles, rbac_api.list_cluster_role_binding(), context_user, "ClusterRole"
)

# Search in RoleBindings for all namespaces
roles = search_and_save_roles(
    roles,
    rbac_api.list_role_binding_for_all_namespaces(),
    context_user,
    "Role",
)

def search_and_save_roles(roles, role_bindings, context_user, role_binding_type):
    for rb in role_bindings:
        if rb.subjects:
            for subject in rb.subjects:
                if subject.kind == "User" and subject.name == context_user:
                    if role_binding_type == "ClusterRole":
                        roles.append(f"{role_binding_type}: {rb.role_ref.name}")
                    elif role_binding_type == "Role":
                        roles.append(
                            f"{role_binding_type} ({rb.metadata.namespace}): {rb.role_ref.name}"
                        )
    return roles

Reply:
Done